src/main/java/org/jabref/gui/edit/ManageKeywordsAction.java

```diff
@@ -1,34 +1,31 @@
 package org.jabref.gui.edit;
 
-import org.jabref.gui.BasePanel;
-import org.jabref.gui.JabRefFrame;
+import org.jabref.gui.StateManager;
 import org.jabref.gui.actions.SimpleCommand;
+import org.jabref.gui.util.BindingsHelper;
 import org.jabref.logic.l10n.Localization;
 
+import static org.jabref.gui.actions.ActionHelper.needsDatabase;
+import static org.jabref.gui.actions.ActionHelper.needsEntriesSelected;
+
 /**
  * An Action for launching keyword managing dialog
  *
  */
 public class ManageKeywordsAction extends SimpleCommand {
 
-    private final JabRefFrame frame;
+    private final StateManager stateManager;
+
+    public ManageKeywordsAction(StateManager stateManager) {
+        this.stateManager = stateManager;
 
-    public ManageKeywordsAction(JabRefFrame frame) {
-        this.frame = frame;
+        this.executable.bind(needsDatabase(stateManager).and(needsEntriesSelected(stateManager)));
+        this.statusMessage.bind(BindingsHelper.ifThenElse(this.executable, "", Localization.lang("Select at least one entry to manage keywords.")));
```

Comment:
How can you bind it if it's read only?

Reply:
The wrapper itself is not readonly, but allows the construction of an readonly property. It's a bit confusing.